default/AI/ProductionAI.py

```diff
@@ -44,7 +35,12 @@ def find_best_designs_this_turn():
     start = time.clock()
     ShipDesignAI.Cache.update_for_new_turn()
     design_cache.clear()
-    design_cache[PriorityType.PRODUCTION_MILITARY] = ShipDesignAI.MilitaryShipDesigner().optimize_design()
+    best_military_stats = ShipDesignAI.MilitaryShipDesigner().optimize_design()
+    best_carrier_stats = ShipDesignAI.CarrierShipDesigner().optimize_design()
+    best_stats = best_military_stats + best_carrier_stats
+    if best_stats:
+        best_stats.sort(key=itemgetter(0))
```

Comment:
Boom!!!

1) reversed required because we need max value first
2) `itemgetter(0)` do same as key=None and can be omited
3) empty list can be sorted too, `if best_stats` can be removed

result should looks like:  `best_stats.sort(reversed=True)`